documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -257,11 +259,24 @@ This will be configurable on `Mice` (i.e. via `InputContext.Mice.ClickConfigurat
 Unlike 1.0 and 2.0, this proposal uses `readonly record struct`s as their only argument for the event action. This allows us to provide more information to the event handlers without breaking in the future. These types are farily simple:
 
 ```cs
-public readonly record struct ButtonEvent<TDevice, TButton>(TDevice Device, TButton Button);
-public readonly record struct ClickEvent<TDevice, TButton>(TDevice Device, TButton Button, Vector2 Position);
-public readonly record struct AxisEvent<TDevice, TAxis>(TDevice Device, int Index, TAxis OldValue, TAxis NewValue);
-public readonly record struct TextEvent<TDevice>(TDevice Device, string? OldText, string? NewText);
-public readonly record struct ConnectionEvent<TDevice>(TDevice Device, bool IsConnected);
+public readonly record struct ConnectionEvent(IInputDevice Device, bool IsConnected);
+public readonly record struct KeyDownEvent(IKeyboard Keyboard, Key Key, bool IsRepeat);
+public readonly record struct KeyUpEvent(IKeyboard Keyboard, Key Key);
+public readonly record struct KeyCharEvent(IKeyboard Keyboard, char Character, int Scancode);
```

Comment:
Looking [at the GLFW input guide](https://www.glfw.org/docs/latest/input_guide.html) it seems `KeyChar` events don't have a scancode, but rather just a "codepoint" (the `char`). KeyDown and KeyUp however do have a scancode.

Reply:
And in 2.X we also have it like this:

```cs
event Action<IKeyboard, Key, int>? KeyDown;
event Action<IKeyboard, Key, int>? KeyUp;
event Action<IKeyboard, char>? KeyChar;
```

Reply:
```suggestion
public readonly record struct KeyDownEvent(IKeyboard Keyboard, Key Key, bool IsRepeat);
public readonly record struct KeyUpEvent(IKeyboard Keyboard, Key Key);
public readonly record struct KeyCharEvent(IKeyboard Keyboard, char Character);
```